Create Translators/Reviewers Table

gaoxk opened the pull request:
![image](https://user-images.githubusercontent.com/15113249/137062401-3898b813-f375-4776-85c8-913c4805c24b.png)

## Notion ticket link

<!-- Please replace with your ticket's URL -->

[Create Translators/Reviewers Table](https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=ba22ac93d09040ff8219c60ac7caabdb&p=4412841e4ef34210968a3aec024b7b07)

<!-- Give a quick summary of the implementation details, provide design justifications if necessary -->

## Implementation description

- used current `users` query for now
- added `UserTable`, flexible with `Translators` and `Reviewers` screens

**note** that
- name column does not sort
- delete user button does nothing
- there is no pagination

Tickets in our task board will be updated accordingly based on what is left over to complete

<!-- What should the reviewer do to verify your changes? Describe expected results and include screenshots when appropriate -->

## Steps to test

1. login and go to /admin
2. view manage translators and manage stories
3. admire the lit af tables 🔥 
4. hover and click a user's name. observe that it takes you to their user page
5. hover and click a user's email. observe that it opens your email client to the user's email
6. observe that the APPROVED LANGUAGES column displays approvedLanguagesTranslation or approvedLanguagesReview based on which admin page tab is selected 

<!-- Draw attention to the substantial parts of your PR or anything you'd like a second opinion on -->

## What should reviewers focus on?

- theres no way to make margins in tables. Convoluted transparent borders are the only way
- sorting by name is not implemented
- delete user button does nothing
- there is no pagination

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

gaoxk commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+
+  const generateBadges = (approvedLanguages: any) =>
+    Object.keys(approvedLanguages).map((apprLang: string) => (
+      // TODO: determine badge color by label
```
Andrew is currently working on utility for this https://github.com/uwblueprint/planet-read/pull/120/files#diff-3c271637fa259c6b8ad6df5ffd2672b04a0dfe32a8c63537964312f6af0ce79bR29

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+      <Td>
+        <IconButton
+          aria-label={`Delete user ${userObj?.firstName} ${userObj?.lastName}`}
+          background="tranparent"
```
nit: should this be `transparent`?

jennchenn replied:
nit `transparent`

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+  const generateBadges = (approvedLanguages: any) =>
+    Object.keys(approvedLanguages).map((apprLang: string) => (
+      // TODO: determine badge color by label
+      <Badge background="orange.50">{`${convertLanguageTitleCase(
```
the badges on the table seem a lot bolder than the badges in the figma mockup (i think especially since each language is in all caps)--is this on purpose?
 
![image](https://user-images.githubusercontent.com/32009013/137339763-fc5adb1a-43e9-4cb5-b352-de8c73f15ff3.png)

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+  const tableBody = users.map((userObj: User, index: number) => (
+    <Tr>
+      <Td>
+        <Link
```
do we want these links to open in a new tab by default?

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+  const tableBody = users.map((userObj: User, index: number) => (
+    <Tr>
+      <Td>
+        <Link
```
the user/email links don't seem to work for me:
![links](https://user-images.githubusercontent.com/32009013/137340014-f734658f-f4fc-46e9-a6a8-5a194ac47bbf.gif)

gaoxk replied:
AHHHH IM A DUMB CUCUMBER AND FORGOT TO PUSH UP CHANGES LOL

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+};
+
+const UsersTable = ({ isTranslators = false, users }: UsersTableProps) => {
+  const approvedLanguanges = users.map((userObj: User) => {
```
nit: typo `approvedLanguanges`->`approvedLanguages`

gaoxk replied:
spelling is not my forte :')

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+
+const UsersTable = ({ isTranslators = false, users }: UsersTableProps) => {
+  const approvedLanguanges = users.map((userObj: User) => {
+    const approvedLanguange = isTranslators
```
nit `approvedLanguange`->`approvedLanguage`

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+      width="95%"
+    >
+      <Thead>
+        <Tr border="1em solid transparent">
```
can the border be added to the bottom of the table as well?
![image](https://user-images.githubusercontent.com/32009013/137340775-26b0aee3-cc26-405e-a490-023d6d31e68e.png)

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+    >
+      <Thead>
+        <Tr border="1em solid transparent">
+          <Th margin="30px">NAME</Th>
```
the name column heading/values don't seem to be aligned (same with the approved languages and the badges)

![image](https://user-images.githubusercontent.com/32009013/137342081-1decb466-baf3-421d-8120-46c8c8914007.png)
![image](https://user-images.githubusercontent.com/32009013/137342118-1c6dd03f-9c2a-4b78-ab51-c6197188fa81.png)

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+      <Td>{generateBadges(approvedLanguanges[index])}</Td>
+      {/* TODO: make delete button functional */}
+      <Td>
+        <IconButton
```
the icon seems a bit off-centre from the heading (but that might just be me
![image](https://user-images.githubusercontent.com/32009013/137341478-d9e11360-0b3f-44a0-b3d0-896378c78e47.png)

gaoxk replied:
hmm I think I could only center it relative to the whole width of the cell, which is larger than the ACTION text, so it still wouldnt be centered. I'm fine with leaving it off to the left a bit - its technically aligned with the front of the ACTION column. LMK what you think!

jennchenn replied:
sounds good to me!

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,89 @@
+import convertLanguageTitleCase from "../../utils/LanguageUtils";
+
+export type UsersTableProps = {
+  isTranslators?: boolean;
```
nit: `isTranslators` sounds weird to me, but i'm not sure if `areTranslators` would be any better
commenting here in case anyone has suggestions for a better name (but also fine with leaving it)

jennchenn reviewed (changes requested):
The table looks very 🔥 🔥 

The links don't seem to work for me at the moment (pasted gif in the comments)

Would it also be possible to prevent the table headers from shifting back/forth when switching between the manage translators/reviewers tabs? (but if it's too much work that's fine not sure how often admin will be switching between the two)
![tab](https://user-images.githubusercontent.com/32009013/137343055-c2b4e8a4-bd46-4196-9c81-c98b1903579a.gif)
)

gaoxk commented:
![image](https://user-images.githubusercontent.com/15113249/137434331-8d952461-4c94-42f1-827a-fa7a6a789b68.png)
🔥 😎 addressed comments,  lmk what y'all think! Thanks for the reviews

jennchenn commented on frontend/src/components/admin/UsersTable.tsx:
```diff
@@ -0,0 +1,99 @@
+
+const UsersTable = ({ isTranslators = false, users }: UsersTableProps) => {
+  const approvedLanguages = users.map((userObj: User) => {
+    const approvedLanguange = isTranslators
```
nit still `approvedLanguange`

gaoxk replied:
HOW DO I KEEP MISSING THESE AHHH
